### ISSUE TITLE
swagger-ui-bundle.js cdn can not work cause the docs can not work.

### ISSUE THREAD

**Issue by zxl777** [tiangolo/fastapi#2735]
/docs
![image](https://user-images.githubusercontent.com/816448/106409931-b2989b00-647c-11eb-8cde-855bf443d153.png)

https://cdn.jsdelivr.net/npm/swagger-ui-dist@3/swagger-ui-bundle.js 
Error!

**Comment by Kludex** [comment]
Duplicated issue. 

We have a PR waiting to be merged already. I'm about to sleep, but feel free to check it. 😗✌️

**Comment by jybaek** [comment]
@Kludex I think it would be nice to write the duplicate issue number. :)

**Comment by zxl777** [comment]
The cdn has worked, and the docs have also worked.
But is it a problem to rely on this CDN?

**Comment by Kludex** [comment]
https://github.com/tiangolo/fastapi/pull/2656

**Comment by omniproc** [comment]
I don't think you should rely on external ressources at all. There are many reasons why this will break. One of course is that the content will simply not be available any longer. But there are also other reasons. In our case - and I imagine this is true for many other companies as well - we can't access that resource because we're behind a proxy that will block access to that CDN content.

Any js dependencies required should be bundled with FastApi and served localy. That way you also keep control on those dependencies that you otherwise can't control.

**Comment by vsatpathy** [comment]
Has anyone been able to resolve this, as it's still not working.

**Comment by Basketballforce** [comment]
You can self-host the required js & css following the steps here:
 https://fastapi.tiangolo.com/advanced/extending-openapi/?h=#self-hosting-javascript-and-css-for-docs

It would be nice if they were hosted by default. I've been having issues as well.

**Comment by vsatpathy** [comment]
@Basketballforce thanks for the link, for the time being I am using the same approach. But would be really helpful if this would be resolved and hosted by default.

**Comment by Pavan-249** [comment]
> I don't think you should rely on external ressources at all. There are many reasons why this will break. One of course is that the content will simply not be available any longer. But there are also other reasons. In our case - and I imagine this is true for many other companies as well - we can't access that resource because we're behind a proxy that will block access to that CDN content.
> 
> Any js dependencies required should be bundled with FastApi and served localy. That way you also keep control on those dependencies that you otherwise can't control.
> 
> I really think this should be flagged as bug since I can't imagine someone would think this is "desired behaviour".

This is the most accurate description. I was in my company network which had blocked my access, as soon as I switched to my private network it loaded up. Thanks for helping out.
